src/core2cuda.rkt

```diff
@@ -0,0 +1,152 @@
+#lang racket
+
+(require generic-flonum)
+(require "imperative.rkt")
+
+(provide cuda-header
+         cuda-supported
+         cuda-reserved
+         core->cuda
+         type->cuda
+         params->cuda      ; required by core2h.rkt
+         set-cuda-header!
+         set-unknown->cuda!)
+
+(define cuda-header (const "#include <fenv.h>\n#include <math.h>\n#include <stdint.h>\n__device__ double e = 2.71828182845904523536;\n\n"))
+(define cuda-supported
+  (supported-list
+    (invert-op-proc (curry set-member? '(array dim size ref for for* tensor tensor*)))
+    fpcore-consts
+    (curry set-member? '(binary32 binary64 binary80 integer))
```

Comment:
I understand you probably just C&P'd the C file and that's fine but I don't think CUDA has 80-bit float ("long double") support?

Reply:
Yeah that's what I did, and ok I just checked it doesn't seem like it does. Removed that.